### PR TITLE
Improve Reminder feature

### LINKED ISSUE #170: Improve reminder feature

### PR DESCRIPTION
**Pull request by justinyjt**
- Disallow reminding for roles with statuses "pending", "rejected" and "complete"
- Update role class to allow checking of whether the role needs reminding
- Update userprefs class to include reminderWindow variable
- Update deadline class to get the reminderWindow from userprefs instead of using default value of 7

### CODE REVIEW

File: src/main/java/seedu/tinner/storage/JsonAdaptedCompany.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public Company toModelType() throws IllegalValueException {
         for (JsonAdaptedRole role : roles) {
             Role currRole = role.toModelType();
             companyRoles.addRole(currRole);
-            if (currRole.getDeadline().isOneWeekAway()) {
+            if (currRole.needsReminding()) {
```
Comment (jinnhl) [inline src/main/java/seedu/tinner/storage/JsonAdaptedCompany.java:124]: Good abstraction to hide the lower-level implementations 👍

File: src/main/java/seedu/tinner/MainApp.java [hunk 2]
```diff
@@ -78,24 +80,24 @@ public void init() throws Exception {
...
                 logger.info("Data file not found. Will be starting with a sample CompanyList");
             }
             initialData = companyListOptional.orElseGet(SampleDataUtil::getSampleCompanyList);
         } catch (DataConversionException e) {
             logger.warning("Data file not in the correct format. Will be starting with an empty CompanyList");
-            reminderList = UniqueReminderList.getReminderList();
             initialData = new CompanyList();
         } catch (IOException e) {
             logger.warning("Problem while reading from the file. Will be starting with an empty CompanyList");
-            reminderList = UniqueReminderList.getReminderList();
```
Comment (jinnhl) [inline src/main/java/seedu/tinner/MainApp.java:95]: Good removal of redundancies :thumbsup:

**Review by jinnhl — APPROVED** [review]
LGTM! Good implementation of SLAP 👍

File: src/main/java/seedu/tinner/model/role/Deadline.java [hunk 3]
```diff
@@ -51,8 +57,8 @@ public static boolean isValidDeadline(String test) {
      *
      * @return true if is within a week from the current date.
      */
-    public boolean isOneWeekAway() {
-        LocalDateTime oneWeekAway = LocalDateTime.now().plusDays(7);
+    public boolean isWithinReminderWindow() {
```
Comment (kerbysoh) [inline src/main/java/seedu/tinner/model/role/Deadline.java:60]: Good renaming of function here for more clarity

File: src/main/java/seedu/tinner/model/role/Role.java [hunk 4]
```diff
@@ -69,6 +69,13 @@ public boolean isSameRole(Role otherRole) {
                 .equals(getName().toString().replaceAll("\\s+", ""));
     }
 
+    /**
+     * Returns true if the role is within the reminder period and status is not pending, rejected or complete
+     */
```
Comment (kerbysoh) [inline src/main/java/seedu/tinner/model/role/Role.java:74]: Nice implementation of new functions to avoid code duplication

**Review by kerbysoh — APPROVED** [review]
LGTM!

**Comment by petermonky** [comment]
Clean code 👍 I'm currently working on UI and changed up the `Reminder` handling logic a bit; will take your changes into account.

**Pull request merged**